sim/battle.ts

```diff
@@ -1136,47 +1143,42 @@ export class Battle extends Dex.ModdedDex {
 
 	getPokemon(id: string | Pokemon) {
 		if (typeof id !== 'string') id = id.id;
-		for (const pokemon of this.p1.pokemon) {
-			if (pokemon.id === id) return pokemon;
-		}
-		for (const pokemon of this.p2.pokemon) {
-			if (pokemon.id === id) return pokemon;
+		for (const side of this.sides) {
+			for (const pokemon of side.pokemon) {
+				if (pokemon.id === id) return pokemon;
+			}
 		}
 		return null;
 	}
 
 	makeRequest(type?: string) {
 		if (type) {
 			this.currentRequest = type;
-			this.p1.clearChoice();
-			this.p2.clearChoice();
+			for (const side of this.sides) {
+				side.clearChoice();
+			}
 		} else {
 			type = this.currentRequest;
 		}
 
 		// default to no request
-		let p1request: any = null;
-		let p2request: any = null;
-		this.p1.currentRequest = '';
-		this.p2.currentRequest = '';
-		let switchTable = [];
+		let requests: any[] = Array(this.sides.length).fill(null);
```

Comment:
```suggestion
		const requests: any[] = Array(this.sides.length).fill(null);
```

You have like 15 more of these.

You can run `npx tslint --project . --fix` to fix them all automatically.